src/bgw/job_stat.c

```diff
@@ -186,30 +189,63 @@ calculate_jitter_percent()
 /* For failures we have standard exponential backoff based on consecutive failures
  * along with a ceiling at schedule_interval * MAX_INTERVALS_BACKOFF */
 static TimestampTz
-calculate_next_start_on_failure(TimestampTz last_finish, int consecutive_failures, BgwJob *job)
+calculate_next_start_on_failure(TimestampTz finish_time, int consecutive_failures, BgwJob *job)
 {
 	float8 jitter = calculate_jitter_percent();
 	/* consecutive failures includes this failure */
 	float8 multiplier = 1 << (consecutive_failures - 1);
+	TimestampTz res;
+	bool res_set = false;
```

Comment:
```suggestion
	volatile bool res_set = false;
```
It's a little subtle, but I think we want `res_set` to be `volatile` with how we're using it. (I think it's legal for the compiler to pull the `res_set = true` before the function calls otherwise, which would leave it in an indeterminate state)